user/token_test.go

```diff
@@ -1,247 +1,326 @@
+		data:   &TokenData{UserId: "12-99-100", IsServer: false, DurationSecs: 3600},
+		config: tokenConfig,
+	}
+
+	token, _ := CreateSessionToken(testData.data, testData.config)
+
+	if token.ID == "" {
+		t.Fatalf("should generate a session token with an ID set")
+	}
+
+	jwtToken, err := jwt.Parse(token.ID, func(t *jwt.Token) (interface{}, error) { return []byte(testData.config.Secret), nil })
+	if err != nil || !jwtToken.Valid {
+		t.Fatalf("should decode and validate the token")
+	}
+
+	claims := jwtToken.Claims.(jwt.MapClaims)
+
+	if int64(claims["dur"].(float64)) != testData.data.DurationSecs {
+		t.Fatalf("token should use the DurationSecs it was given")
+	}
+	if claims["exp"] == nil {
+		t.Fatalf("token expiration should be set")
+	}
+	if claims["iat"] == nil {
+		t.Fatalf("token creation time should be set")
+	}
+	if claims["jti"] == nil {
+		t.Fatalf("token unique id (jti) should be set")
+	}
+
+	td, _ := UnpackSessionTokenAndVerify(token.ID, tokenConfig.Secret)
+
+	if td.DurationSecs != testData.data.DurationSecs {
+		t.Fatalf("we should use the DurationSecs if given")
+	}
+}
+
+func Test_GenerateSessionToken_DurationFromConfig(t *testing.T) {
+
+	testData := tokenTestData{
+		data:   &TokenData{UserId: "12-99-100", IsServer: false, DurationSecs: 0},
+		config: tokenConfig,
+	}
+
+	//given duration seconds trump the configured duration
+	token, _ := CreateSessionToken(testData.data, testData.config)
+
+	if token.ID == "" || token.Time == 0 {
+		t.Fatalf("should generate a session token")
+	}
+
+	td, _ := UnpackSessionTokenAndVerify(token.ID, tokenConfig.Secret)
+
+	if td.DurationSecs != tokenConfig.DurationSecs {
+		t.Fatalf("the duration should be from config")
+	}
+}
+
+func Test_GenerateSessionToken_DurationSecsTrumpConfig(t *testing.T) {
+
+	testData := tokenTestData{
+		data:   &TokenData{UserId: "12-99-100", IsServer: false, DurationSecs: 5},
+		config: tokenConfig,
+	}
+
+	token, _ := CreateSessionToken(testData.data, testData.config)
+
+	if token.ID == "" || token.Time == 0 {
+		t.Fatalf("should generate a session token")
+	}
+
+	td, _ := UnpackSessionTokenAndVerify(token.ID, tokenConfig.Secret)
+
+	if td.DurationSecs != testData.data.DurationSecs {
+		t.Fatalf("the duration should come from the token data")
+	}
+
+}
+
+func Test_GenerateSessionToken_Zendesk_Claims(t *testing.T) {
+
+	testData := tokenTestData{
+		data:   &TokenData{UserId: "12-99-100", IsServer: false, DurationSecs: 5000, Audience: "zendesk"},
+		config: tokenConfig,
+	}
+
+	token, _ := CreateSessionToken(testData.data, testData.config)
+
+	if token.ID == "" || token.Time == 0 {
+		t.Fatalf("should generate a session token")
+	}
+
+	jwtToken, err := jwt.Parse(token.ID, func(t *jwt.Token) (interface{}, error) { return []byte(testData.config.Secret), nil })
+	if err != nil || !jwtToken.Valid {
+		t.Fatalf("should decode and validate the token")
+	}
+
+	claims := jwtToken.Claims.(jwt.MapClaims)
+
+	// check zendesk claims
+	if claims["organization"] != "Patient" {

```

Comment:
should we also check the duration? iat? subject? exp?

Reply:
or it it part of the generic session token?

Reply:
I added this in the generic part